vulkan_generator/struct_factories/generator.py

```diff
@@ -0,0 +1,451 @@
+        remapper_h.write("\n")
+
+        for type_name in vulkan_metadata.types.handle_aliases:
+            remapper_h.write("typedef uint64_t " + type_name + ";\n")
+        remapper_h.write("\n")
+
+        for type_name in vulkan_metadata.types.funcpointers:
+            remapper_h.write("typedef uint64_t " + type_name + ";\n")
+        remapper_h.write("\n")
+
+        for type_name in vulkan_metadata.types.bitmasks:
+            remapper_h.write("typedef uint64_t " + type_name + ";\n")
+        remapper_h.write("\n")
+
+        for type_name in vulkan_metadata.types.bitmask_aliases:
+            remapper_h.write("typedef uint64_t " + type_name + ";\n")
+        remapper_h.write("\n")
+
+        for type_name in vulkan_metadata.types.enums:
+            remapper_h.write("typedef uint64_t " + type_name + ";\n")
+        remapper_h.write("\n")
+
+        for type_name in vulkan_metadata.types.enum_aliases:
+            remapper_h.write("typedef uint64_t " + type_name + ";\n")
+        remapper_h.write("\n")
+
+        for struct in vulkan_metadata.types.structs:
+            remapper_h.write("class " + struct + " {};\n")
+        remapper_h.write("\n")
+
+        all_vulkan_structs = vulkan_metadata.types.structs
+        for struct in sort_structs_dep_order(vulkan_metadata):
+
+            public_members: List[str] = [f"""{struct_factory_name(struct)}();""",
+                                         f"""virtual ~{struct_factory_name(struct)}();"""
+                                         "\n"]
+
+            private_members: List[str] = []
+
+            for member in all_vulkan_structs[struct].members:
+                member_data = all_vulkan_structs[struct].members[member]
+                public_members += [generate_struct_member_setter(member_data, vulkan_metadata),
+                                   generate_struct_member_getter(member_data, vulkan_metadata)]
+                private_members += [generate_struct_member(member_data, vulkan_metadata)]
+
+            public_members += ["",
+                               "size_t VkStructMemorySize() override;",
+                               "size_t PNextChainMemorySize() override;",
+                               "",
+                               f"""{struct} Generate();"""]
+
+            class_def = codegen.create_class_definition(struct_factory_name(struct),
+                                                        public_inheritance=["VkStructFactory"],
+                                                        public_members=public_members,
+                                                        private_members=private_members)
+
+            remapper_h.write(class_def)
+            remapper_h.write("\n")
+
+        remapper_h.write(dedent("""
+            }
+            }
+
+        """))
+
+def generate_factory_ctor_def(struct : str) -> str :
+    return dedent(f"""
+                  {struct_factory_name(struct)}::{struct_factory_name(struct)}() {{
```

Comment:
Personal Opinion: 

I believe using this as 

`{struct}Factory::{struct}Factory` is actually shorter and more readable. It's a bit problem if we want to change it from Factory to something else but having a function name three times longer then what we actually append seems a bit overkill.